### DISCUSSION TITLE
radio box default from cli or configuration file

### DISCUSSION THREAD

**Discussion by chdelfosse** [epeios-q37/atlas-python#10] [General]
Since getValue(id), applied to a radio box, returns the value of the "check" ed option, one could think (by orthogonality) that setValue
would pre-"check" an option. This is not the case.
I found two workarounds

- have Main.html contain dynamic strings, instantiated at run time, only one of which is "checked" 
- build the complete set of radio buttons using createHTML
Any thoughts on this?

**Comment by epeios-q37** [comment]
Thanks for starting this discussion.

I've just tested it: actually, `set_value(<id>, True)` should work… If this is not the case, could you open an issue with an example of code?

Note that there is another way to retrieve which one of a group of radio buttons is checked, instead of using `get_values(…)` with the ids of the radio buttons, which is somewhat cumbersome. You can use the `xdh:radio` attribute as explained here: https://atlastk.org/api/en/#values. An example can be found by selecting the *radio* widget of the *Widgets* demo available here: https://q37.info/s/vwpsw73v (look at the *Submit* button).  
With this method, `get_value(…)` returns the value (i.e. the content of its `value` attribute) of the checked radio button. What I have in mind is the use `set_value(…)`, in combination with the same `xdh:radio`, where you give the value of the radio button you want being checked. Radio buttons so no more need to have an id.

**Reply by chdelfosse** [reply]
I use "xdh:radio" feature  and I confirm that it is very convenient. 
I will dig into older revisions of my code to find the test that failed (?) and led me to try and find workarounds for setting a default choice on a set of radio buttons.
Or confirm your test and ask you to close the issue, whichever comes first.

**Comment by chdelfosse** [comment]
I confirm.

set_value(<id>,True) [python] works indeed and is a LOT simpler and more convenient than any of my workarounds.
It is the only reason to assign an id to individual radio buttons (changing the default value at any time by
software rather than user interaction).

my code had : 
set_value(<id>,"checked")
and failed.

As an afterthought, I tried
set_attribute(<id>,"checked",True)
and it works as well, perhaps even more logically

Thanks for your answer

**Comment by epeios-q37** [comment]
The backend has been updated.
It is now possible to check a radio button with a `set_value(…)` on an element with an `xdh:radio` attribute. To uncheck all the radio buttons of a group, you just have to pass an empty value.
In the same way, you can now select an entry of a list box by calling a `set_value(…)` on the `select` element corresponding to this entry, with the value of the `value` attribute of the entry (the `option` element).
The *radio* widget of the *Widgets* demo available at https://q37.info/s/vwpsw73v has been updated to use this new functionalities.
